Extend RTN-005 to include Budget Planning

drphilmarshall opened the pull request:
This PR tracks changes proposed to extend RTN-005 to cover our budget planning process and tools, addressing the following tickets:

* [PREOPS-1078](https://jira.lsstcorp.org/browse/PREOPS-1078): Document the Rubin Operations Planning Tools in RTN-005
* [PREOPS-3177](https://jira.lsstcorp.org/browse/PREOPS-3177): Extend RTN-005 to include budget planning ("scrub") process

As a reminder, here are the two JOR22 review recommendations that this RTN-005 extension will address:

* _"Define and formalize an appropriate bottom up process to review and scrub budgets and performance, review and approve requests beyond baseline targets, and to prioritize the annual scope of the program by end of FY22."_ ([PREOPS-1131](https://jira.lsstcorp.org/browse/PREOPS-1131))

*  _"Implement a process for developing a consolidated, detailed, bottom-up, cost estimate informed by the schedule milestones utilizing integration of cost estimate and schedule tools. By the next annual review."_ ([PREOPS-1127](https://jira.lsstcorp.org/browse/PREOPS-1127))

I am interpreting these as "write down a budget scrubbing and work planning process for Rubin ops, put it up for joint ops review in February 2023, and then execute it prior to the start of FY24."

I will ping @ranpalgill , @cathypetry, @rblum5 and Christine as reviewers when the new version is ready; for now, you can use this PR to track the changes as I make them, and contribute either by pushing commits directly to this branch, or better, making a sub-branch ("u/your-name/xbudget") and submitting a PR against the primary dev branch ("u/pjm/xbudget"). 

Then, I'll ask the ADs to review the changes, and give feedback on the proposed process.

drphilmarshall commented:
@rblum5 @cathypetry @ranpalgill : My updates to RTN-005 are ready for review. Would you mind please taking a look? Suggested changes in the [Files changed](https://github.com/lsst/rtn-005/pull/20/files) tab are most welcome. You can also read the compiled PDF document at https://rtn-005.lsst.io/v/u-pjm-xbudget .

Once we think the sandboxing/scrub process sounds OK, I'll ask the ADs to review.

Thanks!

rblum5 commented on body.tex:
```diff
@@ -692,7 +699,95 @@ \subsection{Reporting Cycle}
+DOE HEP holds additional reserves centrally, which can also be used as both contingency and management reserve.
+NSF AST allows line item budget over-estimation by up to 10%, to allow for increases in cost.
+These generate carry-forward (along with any other under-spending that may occur), which we use as contingency reserve.
+NSF also holds reserves of both kinds centrally, to cover high cost impact realized risks as well as unexpected events.
```
I would say only that "NSF holds management reserve to handle unexpected events."

drphilmarshall replied:
Happy to make that change - but is this really true? What does NSF do about registered risks (known unknowns) that have high cost exposure, eg that could not be paid for by the accumulated 10% allowances / carry-forward? I was assuming that it would use its centrally-held reserves for such risks if they are realized - which is to say that their reserves are both "management" (unexpected events) and "contingency" (known risks that are unfortunately realized).

rblum5 commented on body.tex:
```diff
@@ -692,7 +699,95 @@ \subsection{Reporting Cycle}
+NSF also holds reserves of both kinds centrally, to cover high cost impact realized risks as well as unexpected events.
+
+The cost impact analysis of the Rubin Operations risk register informs the level of contingency reserve needed.
+Residual (post-mitigation) cost exposure for each risk is computed by taking the  product of the residual probability of the risk being realized (in any given year) and the likely annual cost of addressing that risk.
```
maybe add "While Rubin holds a number of potentially very costly risks, these are generally very unlikely to occur."

drphilmarshall replied:
Good. I looked at the risk analysis: it's the "significant" risks that carry most of the total cost exposure, the "damaging/major" and "catastrophic/extreme" risks don't carry much exposure at all (because their likelihood is so low, thankfully).

rblum5 commented on body.tex:
```diff
@@ -692,7 +699,95 @@ \subsection{Reporting Cycle}
+Included in the risk register is the risk of federal funding allocations being reduced, with a likely cost (if realized in any given year) of some \$1.5M per operations partner.
+(At SLAC, for example, this would cover about 7\% of the operating expense -- enough to pay the labor bills during a 1 month funding allocation delay.)
+
+Reserve maintenance is accounted for in our federal budget requests, either as an explicit difference between cost and budget on the DOE side, or as line by line allowances for increases in cost on the NSF side.
```
I think this is the ideal. We are not there yet on NSF side in our estimations.

drphilmarshall replied:
No - but once this is reviewed positively it can be our MO for the next 5-year proposal.

rblum5 commented on body.tex:
```diff
@@ -692,7 +699,95 @@ \subsection{Reporting Cycle}
+(At SLAC, for example, this would cover about 7\% of the operating expense -- enough to pay the labor bills during a 1 month funding allocation delay.)
+
+Reserve maintenance is accounted for in our federal budget requests, either as an explicit difference between cost and budget on the DOE side, or as line by line allowances for increases in cost on the NSF side.
+At both operations partners the reserve is maintained as carry-forward from one year's spending to the next, with it being spent down to zero either during the post-operations period (at SLAC), or at the end of each 5-year renewal period at NOIRLab.
```
I would revise as "At both operations partners carry forward is maintained from one year's plan to the next, with it being spent down to zero either during the post-operations period (at SLAC), or before the end of each 5-year renewal period at NOIRLab. Each year a plan is described in the POP for how carry forward is allocated."

drphilmarshall replied:
OK, thanks!

rblum5 commented on body.tex:
```diff
@@ -692,7 +699,95 @@ \subsection{Reporting Cycle}
+Reserve maintenance is accounted for in our federal budget requests, either as an explicit difference between cost and budget on the DOE side, or as line by line allowances for increases in cost on the NSF side.
+At both operations partners the reserve is maintained as carry-forward from one year's spending to the next, with it being spent down to zero either during the post-operations period (at SLAC), or at the end of each 5-year renewal period at NOIRLab.
+(The restriction on carrying forward funds between renewal periods means that the level of reserve held at NOIRLab varies significantly over the course of the 5 years.
+Risks that are realized early in the 5-year period are more likely to need supplemental funding from MSF's centrally-held reserves than those that are realized later on, when the carry-forward has been built up.
```
MSF->NSF

drphilmarshall replied:
Good spot, thank you

rblum5 commented on body.tex:
```diff
@@ -692,7 +699,95 @@ \subsection{Reporting Cycle}
+Risks that are realized early in the 5-year period are more likely to need supplemental funding from MSF's centrally-held reserves than those that are realized later on, when the carry-forward has been built up.
+It may be possible to purposefully hold back scope so as to build up the carry-forward quickly at the start of a renewal period.)
+
+Requests beyond target (RBTs) are typically made by the departments in order to address risks as they are realized.
```
"...address risks and/or opportunities as ..."

We probably need more on opportunities generally. But can wait for later.

drphilmarshall replied:
Modification made. Need to think about opportunity registration, agreed.

rblum5 commented on body.tex:
```diff
@@ -692,7 +699,95 @@ \subsection{Reporting Cycle}
+It may be possible to purposefully hold back scope so as to build up the carry-forward quickly at the start of a renewal period.)
+
+Requests beyond target (RBTs) are typically made by the departments in order to address risks as they are realized.
+(They may also be made in order to take advantage of opportunities arising.)
```
suggest delete if you accept the change above.

drphilmarshall replied:
Done

drphilmarshall commented on body.tex:
```diff
@@ -692,7 +699,95 @@ \subsection{Reporting Cycle}
+Rubin Operations' two funding agencies offer differing guidelines on how reserve should be managed.
+DOE allows reserves to be built up and carried forward at SLAC, and used as both contingency and management reserve.
+DOE HEP holds additional reserves centrally, which can also be used as both contingency and management reserve.
+NSF AST allows line item budget over-estimation by up to 10%, to allow for increases in cost.
```
```suggestion
NSF AST allows line item budget over-estimation by up to 10\%, to allow for increases in cost.
```

drphilmarshall commented on body.tex:
```diff
@@ -692,7 +699,95 @@ \subsection{Reporting Cycle}
+Residual (post-mitigation) cost exposure for each risk is computed by taking the  product of the residual probability of the risk being realized (in any given year) and the likely annual cost of addressing that risk.
+The total cost exposure is then the sum, over all risks in the register, of these products.
+
+Currently, the total cost exposure, and hence the needed level of contingency reserve, is approximately 15\% of the total annual operating budget at each operations partner.
```
```suggestion
Currently, the total cost exposure, and hence the needed level of contingency reserve, is expected to be approximately 15\% of the total annual operating budget at each operations partner.
```

drphilmarshall commented on body.tex:
```diff
@@ -1,25 +1,26 @@
 
-The annual planning process starts by setting high level milestones for the year, which are centered around releasing data to the public and major maintenance to the telescope system once we enter the phase of full survey operations.   
-In a six-month cycle, the Leadership Team builds a series of milestone activities that are discrete pieces of work within Departments and Teams to collectively deliver the high level milestones.  
+The annual planning process starts by setting high level milestones for the year, which are centered around releasing data to the public and major maintenance to the telescope system once we enter the phase of full survey operations.
```
Nice! Adopted.

drphilmarshall commented on body.tex:
```diff
@@ -1,25 +1,26 @@
-The annual planning process starts by setting high level milestones for the year, which are centered around releasing data to the public and major maintenance to the telescope system once we enter the phase of full survey operations.   
-In a six-month cycle, the Leadership Team builds a series of milestone activities that are discrete pieces of work within Departments and Teams to collectively deliver the high level milestones.  
+The annual planning process starts by setting high level milestones for the year, which are centered around releasing data to the public and major maintenance to the telescope system once we enter the phase of full survey operations.
+In a six-month cycle, the Leadership Team builds a series of milestone activities that are discrete pieces of work within Departments and Teams to collectively deliver the high level milestones.
```
Good idea, done.

drphilmarshall commented on body.tex:
```diff
@@ -692,7 +699,95 @@ \subsection{Reporting Cycle}
 
+Typically, each team works up its part of the sandbox mostly independently, and then presents it to the department's management team around mid-May in a sandbox pitch session.
+(A template slide deck for each team's pitch is provided by the Director's Office, to ensure that all aspects of the sandboxing exercise are addressed.)
+The teams then address the feedback they have been given by the management team, and then present their sandboxes to the other teams in the department and also the Directorate and Program Operations teams from the Director's Office at the sandbox workshop at the end of the month.
```
I'll clarify in the text!

drphilmarshall commented on body.tex:
```diff
@@ -692,7 +699,95 @@ \subsection{Reporting Cycle}
+Typically, each team works up its part of the sandbox mostly independently, and then presents it to the department's management team around mid-May in a sandbox pitch session.
+(A template slide deck for each team's pitch is provided by the Director's Office, to ensure that all aspects of the sandboxing exercise are addressed.)
+The teams then address the feedback they have been given by the management team, and then present their sandboxes to the other teams in the department and also the Directorate and Program Operations teams from the Director's Office at the sandbox workshop at the end of the month.
+This sandbox workshop at the end of May / beginning of June gives all the departments, including the Director's Office, a chance to review, and probe, the ground-up planning in each team.
```
Yes: needs some design thinking on how much time it will need, and how to best allocate that time. Perhaps a sequence of daily zoom sessions.

drphilmarshall commented on body.tex:
```diff
@@ -692,7 +699,95 @@ \subsection{Reporting Cycle}
+The changes to the spending plan, once captured in the Rubin planning tools, are implemented in NOIRLab's and SLAC's lab-wide budget models in July.
+Statements of work for the next fiscal year's contracts and MPOs are straightforwardly derived from the scrubbed plans, for their July deadlines.
+
+The teams then plan work against their minor milestones in September, incorporating any community input they may have collected at the annual project and community workshop (PCW) in August.
```
Yes - will clarify in the text

womullan reviewed (approved):
ok - I have added glossary terms but miss RAS and NICRA

ranpalgill commented on body.tex:
```diff
@@ -1,25 +1,27 @@
-The annual planning process starts by setting high level milestones for the year, which are centered around releasing data to the public and major maintenance to the telescope system once we enter the phase of full survey operations.   
-In a six-month cycle, the Leadership Team builds a series of milestone activities that are discrete pieces of work within Departments and Teams to collectively deliver the high level milestones.  
+The annual planning process starts by reviewing, revising, and adding major milestones to the next fiscal year plan.
+These are centered around releasing data to the public and major maintenance to the telescope system once we enter the phase of full survey operations, and its these major milestones that are tracked and reported on in the annual NOIRLab Program Operations Plan (POP) and SLAC Field Work Proposal (FWP).
```
```suggestion
These are centered around releasing data to the public and major maintenance to the telescope system once we enter the phase of full survey operations, and it's these major milestones that are tracked and reported on in the annual NOIRLab Program Operations Plan (POP) and SLAC Field Work Proposal (FWP).
```

drphilmarshall replied:
Good catch, thanks!

ranpalgill commented on body.tex:
```diff
@@ -46,7 +48,7 @@ \subsection{Annual Reporting}
 The FWP is delivered in June of the current fiscal year, and covers the federal budget request for the next two fiscal years.
 The FWP is previewed to SLAC management and DOE in February in advance of the final delivery in June.
-Because the POP for NSF and FWP for DOE are out of phase, Rubin does high level planning in early Q2 of the financial year (January and February).
+Because the POP for NSF lags the FWP for DOE by one year, and their submission dates differ by several months, Rubin does high level planning in early Q2 of the financial year (January and February) and plans 2 years ahead, as required for the SLAC FWP.
```
```suggestion
Because the POP for NSF lags the FWP for DOE by one year, and their submission dates differ by several months, Rubin does high level planning in early Q2 of the fiscal year (January and February) and plans 2 years ahead, as required for the SLAC FWP.
```

drphilmarshall replied:
Another one, thanks!

ranpalgill commented on body.tex:
```diff
@@ -373,14 +376,15 @@ \subsection{Defining the Schedule with Milestones}
 %\item The \texttt{Start Date} is the date when the work for the milestone should begin.
 %  This is the date that the Smartsheet will use in a visual comparison between the fraction of work completed and the fraction of time elapsed, to help track progress on the epics.
-\item In the \texttt{Linked Issues} field, create a \textit{``blocks''} link to the parent milestone to reflect how that milestone contributes to acheiving that milestone. 
+\item In the \texttt{Linked Issues} field, create a \textit{``blocks''} link to the parent milestone to reflect how that milestone contributes to acheiving that milestone.
```
```suggestion
\item In the \texttt{Linked Issues} field, create a \textit{``blocks''} link to the parent milestone to reflect how that milestone contributes to achieving that milestone.
```

drphilmarshall replied:
+1

ranpalgill commented on body.tex:
```diff
@@ -414,7 +418,7 @@ \subsubsection{Defining Epics} \label{sec:planning-epics}
 \item The \texttt{Component} field set to the appropriate Department;
 \item The \texttt{Story Points} field set to a (non-zero) estimate of the effort required to complete the \gls{epic} in terms of \glspl{SP} (see \S\ref{sec:effort}).
-\item In the \texttt{Linked Issues} field, create a \textit{``blocks''} link to the parent milestone to reflect how that milestone contributes to acheiving that milestone. 
+\item In the \texttt{Linked Issues} field, create a \textit{``blocks''} link to the parent milestone to reflect how that milestone contributes to acheiving that milestone.
```
```suggestion
\item In the \texttt{Linked Issues} field, create a \textit{``blocks''} link to the parent milestone to reflect how that milestone contributes to achieving that milestone.
```

drphilmarshall replied:
thanks!

ranpalgill commented on body.tex:
```diff
@@ -692,7 +700,98 @@ \subsection{Reporting Cycle}
+(This coordination may be facilitated by the department's management team, e.g.\ following the pitch sessions.)
+
+Following the Sandbox Workshop, the Director's Office adjusts and finalizes the overall spending plan by the end of June, in time for the submission of the NOIRLab POP, and SLAC's DOE budget briefing.
+This June period of post-sandbox spending plan development and cost calculation involves the Directors Office staff making updates to the planning tools (Appendix~\ref{sec:planning-tools}), with review by teh departmental management teams.
```
```suggestion
This June period of post-sandbox spending plan development and cost calculation involves the Directors Office staff making updates to the planning tools (Appendix~\ref{sec:planning-tools}), with review by the departmental management teams.
```

ranpalgill commented on body.tex:
```diff
@@ -692,7 +700,98 @@ \subsection{Reporting Cycle}
+The ground-up spending plan developed during the annual cycle is to cover operations work defined by the milestone activities, as well as any regular work captured in bucket epics.
+This work includes mitigation of risks held by the teams, as described in the response plans in the risk register.
+Any risks that are realized during the year will require action that is not budgeted for.
+Additional spending to address realized risks is enabled by Rubin's request beyond target (RBT) process, whereby reserve funds are released for use by the departments in response to a specific mid-year request made in a custom Jira ticket.
```
I think RBT process should be captured in the CCB process - that might be a specific JIRA ticket but it would be one related to the CCB process

drphilmarshall replied:
Yes - let's remember to point this out in the text when we do the CCB update.

ranpalgill commented on body.tex:
```diff
@@ -692,7 +700,98 @@ \subsection{Reporting Cycle}
+(At NOIRLab, carry-forward use must be declared a year in advance in the POP: this is why the sandboxing process includes a request that the operations teams pitch RBTs for the coming year at that stage.)
+The RBT itself is a Jira ticket submitted by a department's management team on behalf of the team that is addressing the risk.
+Review is by the Directorate team, as budget holders.
+The spending associated with an approved RBT is not captured in a change to the labor or non-labor plan (which together define the baseline spending plan), but instead is tracked as a variance relative to those plans.
```
ah so maybe my comment about CCB for RBT is no relevant. I don't think i fully understand the RBT process, does it start proposed as a JIRA ticket and only if granted goes into the CCB process or something like that? - just feel it needs clarification

drphilmarshall replied:
Sure. This is vague because I am attempting to design a specific change process in the absence of the more general change process. So I am hazarding a guess that it will make most sense for teh RBT itself to be a Jira ticket (what else could it be?) and that its processing will be CCB-like. I think we need to establish the CCB process after the JOR and before May, so that we are set up to accept and process RBTs in June.

drphilmarshall commented on body.tex:
```diff
@@ -226,64 +229,64 @@ \subsubsection{Team Leads and other Leadership Roles}
+      \item Update lab-wide budget models (RAS) from Staffing and Nonlabor plans, for quarterly forecasting.
       \item Semi-annual process of planning epics and L3s for next six months
-      \item Planning for 2nd half may trigger L1-L2 milestone changes needed in next FY
+      \item Planning for 2nd half may trigger major milestone changes needed in next FY:
```
I think we really did mean "may" and not "May", here - resolving.

drphilmarshall commented on body.tex:
```diff
@@ -226,64 +229,64 @@ \subsubsection{Team Leads and other Leadership Roles}
    \begin{itemize}
-      \item Update budget tools with any major changes for NICRA proposal (Will have already updated RAS from Staffing Plan and Nonlabor for quarterly forecasting) in December
+      \item Update lab-wide budget models (RAS) from Staffing and Nonlabor plans, for quarterly forecasting (December, March).
+      \item Update budget tools with any major changes, for AURA NICRA proposal. Incorporate any changes in SLAC overhead rates and escalations.
```
Now addressed, thanks Wil.

drphilmarshall commented on body.tex:
```diff
@@ -692,7 +700,98 @@ \subsection{Reporting Cycle}
+
+Figure~\ref{fig:annual-cycle} illustrates the phases of Rubin Operations budget development and spending planning throughout the year, as laid out in Section~\ref{sec:annual-planning-timeline}.
+In April, following several months collecting high level and external budget information and implementing changes requested during the previous summer, the Director's Office sets major (Level 1 and some Level 2) milestones, and issues departmental budget guidance for the next two fiscal years.
+(Ideally, the Joint agency Operations status Review (JOR) would be held in early April, to provide external expert oversight of the plan.)
```
Thanks @womullan - will do this from now on with all acronyms.

drphilmarshall commented on body.tex:
```diff
@@ -692,7 +700,98 @@ \subsection{Reporting Cycle}
+Each department is given a ``sandbox'' workbook to work up during this period.
+This sandbox is initialized with the department's labor and non-labor plans (that were used in setting the budget), the current staffing plan, the progress against milestones in the current fiscal year, and the major milestones for the coming fiscal year.
+They are then invited to ``scrub'' their plans, by working this sandbox up with analysis of recent performance, and any changes they want to propose for the next fiscal year, based on the minor milestones and associated activities they want to prioritize.
+They are also invited to scrub their part of the risk register, update their risk response plans in concert with their planned activities, and anticipate any requests beyond target they may need to make.
```
Good catch: I think the "theys" are justified here, because "we" in this document refers to Rubin Ops L0 and L1 mgmt plus program ops (ie the coauthors of the doc), not L2 mgmt (Team Leaders) more generally. However, the passive voice is mismatched with the rest of the document. I will fix that.

drphilmarshall commented on body.tex:
```diff
@@ -226,64 +229,64 @@ \subsubsection{Team Leads and other Leadership Roles}
    \begin{itemize}
-      \item Update RAS from Staffing Plan (NOIRLab matrix leveling)
-      \item Update Nonlabor for quarterly forecasting
+      \item Update lab-wide budget models (RAS) from Staffing and Nonlabor plans, for quarterly forecasting.
```
This is now resolved - thanks @womullan !

drphilmarshall commented:
Thanks all! I just addressed all comments will merge this PR. The only comment I punted on was Ranpal’s query about the RBT process. The problem is that we can only design this vaguely because the more general, needed, Change Control process is not yet in place and I think we do want the RBT process to be consistent with it and close to it in terms of how the process works and feels. If we think we need to work up this aspect of RTN-005 in more detail before the review we can start a new branch/PR - and then we'll come back to Change Control / RBT process in March in any case.

Likewise, if you didn't get a chance to review this document in time, please feel free to browse the PR and then submit a Jira ticket with the proposed changes (and of course, a PR of your own would be most welcome).
